qiskit/qpy/binary_io/value.py

```diff
@@ -202,14 +306,101 @@ def _read_parameter_expression_v3(file_obj, vectors):
             raise exceptions.QpyError("Invalid parameter expression map type: %s" % elem_key)
         symbol_map[symbol] = value
 
-    return ParameterExpression(symbol_map, expr)
+    return ParameterExpression(symbol_map, expr_)
+
+
+def _read_expr(
+    file_obj,
+    clbits: collections.abc.Sequence[Clbit],
+    cregs: collections.abc.Mapping[str, ClassicalRegister],
+) -> expr.Expr:
+    # pylint: disable=too-many-return-statements
+    type_key = file_obj.read(1)
```

Comment:
Do you think we should use a size variable here just for future proofing? The type key is always 1 byte now (and probably for the foreseeable future) but it just feels like making this something like `EXPR_TYPE_KEY_SIZE` would be a bit more maintainable longer term and maybe easier to read (although this is debatable because I knew the intent of the magic number here at least).

Reply:
I can't foresee a situation where there are more than 256 different `Expr` nodes, but because I've just written that, I've basically guaranteed that it'll happen one day.

I've factored out all the magic ones in 11e2eaa1a.